apis/s3/src/main/java/org/jclouds/s3/options/PutObjectOptions.java

```diff
@@ -82,6 +82,11 @@ public PutObjectOptions withAcl(CannedAccessPolicy acl) {
          this.replaceHeader(CANNED_ACL, acl.toString());
       return this;
    }
+   
+   public PutObjectOptions withServerSideEncryption() {
+       this.replaceHeader("x-amz-server-side-encryption", "AES256");
+       return this;
+   }
```

Comment:
This should be in our code.  The specific option needs to be created and passed through to the putBlob method using the overloaded method.